### PR TITLE
MAINT: Add a base of CONTRIBUTING guidelines

### LINKED ISSUE #12: Telecon, September 24th, 1 PM PST
Following up on https://github.com/nipreps/dmriprep/pull/8#issuecomment-531207953, according to the [poll](https://www.when2meet.com/?8108971-Y8OeY), 9/24 at 1 PM PST seems to work for everyone who has filled in their availability so far. Barring any conflicts that arise (please pipe up here), let's set that as the time for our telecon. I have set up a zoom call for this with the following details: 

Join Zoom Meeting: https://washington.zoom.us/j/701882231

One tap mobile
+16699006833,,701882231# US (San Jose)
+16468769923,,701882231# US (New York)

Dial by your location
        +1 669 900 6833 US (San Jose)
        +1 646 876 9923 US (New York)
Meeting ID: 701 882 231
Find your local number: https://zoom.us/u/adJpEwdYeg

Join by SIP
701882231@zoomcrc.com

Join by H.323
162.255.37.11 (US West)
162.255.36.11 (US East)
221.122.88.195 (China)
115.114.131.7 (India)
213.19.144.110 (EMEA)
103.122.166.55 (Australia)
209.9.211.110 (Hong Kong)
64.211.144.160 (Brazil)
69.174.57.160 (Canada)
207.226.132.110 (Japan)
Meeting ID: 701 882 231

### PR DESCRIPTION
**Pull request by oesteban**
To be discussed in tomorrow's telecon (Close #12)

There are some descriptions I'll be working on today and tomorrow
morning before finishing the proposal.

This is related (sideways) to #7 too.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+to diffusion MRI data.
+That means *dMRIPrep* shares with *fMRIPrep* (and more generally, other
+*NIPreps* [NeuroImaging Preprocessing pipelines]) the fundamental design principles:
+
+  1. The tool only and fully supports BIDS and BIDS-Derivatives for the input and output data.
+  1. The tool is packaged as a fully-compliant [BIDS-App][link_bidsapps], not just in its user
+     interface, but also in the continuous integration, testing and delivery.
+  1. The tool is rigorously restricted to diffusion MRI preprocessing tasks, including:
+     head-motion correction, eddy-currents-derived artifact correction, susceptibility-distortion
+     correction, denoising, B1-bias field correction, session-drift correction, co-registration
```
[the remark this answers is not in the document]
Comment (arokem) [reply CONTRIBUTING.md]: Should this be a discussion we have here, or should we take it separately?

**Comment by richford** [comment]
LGTM. The only minor change I'd recommend is the inclusion of the "[imposter syndrome disclaimer](https://github.com/nipy/dmriprep/blob/master/README.md#contributing)" from the nipy/dmriprep repo. I think the extra language can help to further foster inclusion. If the disclaimer is contentious, I'm happy to move that to another issue so that we can have that discussion without holding up this PR.

I agree with the "recognizing contributions" section and I tend to lean more towards maximal authorship inclusion (though my [physics bias might be showing here](https://www.nature.com/news/physics-paper-sets-record-with-more-than-5-000-authors-1.17567)).

**Comment by oesteban** [comment]
Would love to hear opinions from @josephmje and @dpys

**Comment by josephmje** [comment]
I think this looks fantastic! I saw the contributing guide from tedana and noticed that this is inspired by the same place. Some additional thoughts:

1. There was talk about running black and potentially a developer setup? Is that still the case? If so, that should be mentioned here.
2. Including a blurb on the Style Guide and writing in ReStructuredText for the docstrings.

**Comment by dPys** [comment]
I agree with the contributing guidelines as outlined. Some notes: 
*The numbering of principles seem to all be labeled as '1.' -- I'm guessing this is a typo?
*Should we mention 'parsimony' as a general principle? i.e. if a purely-pythonic tool is available to a third-party equivalent, all else being equal, then pure-python is preferred to minimize external dependencies. e.g. preference for dipy's gibbs suppression over and above mrtrix's mrdegibbs which does the equivalent but demands mrtrix be added as a dependency. Now if it were the case that mrtrix was already a dependency, and mrdegibbs was a faster option or consumed less memory resources, then that would require different consideration. But I think the overarching principle would be that parsimony, whenever possible, is preferred. And that wouldn't just be a dmriprep principle, since it is fundamentally a pythonic one.

Curious to hear thoughts on this,
@dPys

**Comment by oesteban** [comment]
> *The numbering of principles seem to all be labeled as '1.' -- I'm guessing this is a typo?

Nope, that's the formatting recommendation of markdown - it will reassign numbers when rendering.

> *Should we mention 'parsimony' as a general principle? i.e. if a purely-pythonic tool is available to replace a third-party equivalent, all else being equal, then pure-python is preferred to minimize external dependencies. e.g. preference for dipy's gibbs suppression over and above mrtrix's mrdegibbs which does the equivalent but demands mrtrix be added as a dependency. Now if it were the case that mrtrix was already a dependency, and mrdegibbs was a faster option or consumed less memory resources, then that would require different consideration. But I think the overarching principle would be that parsimony, whenever possible, is preferred. And that wouldn't just be a dmriprep principle, since it is fundamentally a pythonic one.

I agree with this principle in general (not just for dMRIPrep), so for me there is no need to have it in writing. I'm happy including some of the language you proposed if others think we have to make this explicit.

**Pull request merged**